packages/autocannon-ui-frontend/src/components/App.jsx

```diff
@@ -33,32 +34,63 @@ const useStyles = makeStyles(theme => ({
   },
   resultsGridItem: {
     marginTop: theme.spacing(2)
+  },
+  buttonsContainer: {
+    gap: '1rem'
   }
 }))
 
 function App() {
   const classes = useStyles()
 
-  const [results, setResults] = useState([])
+  const [results, setResults] = React.useState([])
+  const [selectedResults, setSelectedResults] = React.useState([])
+  const [isOpenCompareDialog, setIsOpenCompareDialog] = React.useState(false)
 
-  function resultsReceivedHandler(resultSet) {
+  const handleOnReceivedResults = React.useCallback(resultSet => {
     setResults(results => [...results, resultSet])
-  }
+  }, [])
+
+  const handleOnSelectResultSet = React.useCallback((resultSet, isSelected) => {
+    if (isSelected) {
+      setSelectedResults(results => [...results, resultSet])
+    } else {
+      setSelectedResults(results =>
+        results.filter(
+          item => item.start !== resultSet.start || item.url !== resultSet.url
+        )
```

Comment:
do we have to apply logic here? can't we assume the in memory reference is the same?

Reply:
No, we don't have to apply the logic.